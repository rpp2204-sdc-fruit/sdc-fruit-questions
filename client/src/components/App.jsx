import axios from 'axios';
import React, { useState, useEffect } from 'react';

import { useParams } from 'react-router-dom';
import Overview from './overview_module/Overview.jsx';
import QandAModule from './questions_answers_module/QandAModule.jsx';
import ReviewsModule from './reviews_module/ReviewsModule.jsx';
import ProductLinks from './ProductLinks.jsx';
import Topbar from './Topbar.jsx';

function App() {
  const { id } = useParams();
  const [products, setProducts] = useState([]);
  const [productId, setProductId] = useState(0);
  const [productName, setProductName] = useState('');

  useEffect(() => {
    axios.get(`/products/${id}`).then((res) => {
      setProductName(res.data.name);
      setProductId(res.data.id);
    });
  }, [id]);

  useEffect(() => {
    const options = {
      method: 'get',
      url: `/products`,
    };

    axios(options)
      .then((response) => {
        const productList = response.data;
        setProducts([...productList]);
      })
      .catch((error) => {
        console.log(error);
      });
  }, []);

  const widgets =
    productId !== 0 ? (
      <>
        <Overview product_id={productId} />
        <QandAModule product_id={productId} product_name={productName} />
        <ReviewsModule product_id={productId} product_name={productName} />
      </>
    ) : null;

  return (
    <div>
      <Topbar />
      {widgets}
      <ProductLinks products={products} />
    </div>
  );
}

export default App;
