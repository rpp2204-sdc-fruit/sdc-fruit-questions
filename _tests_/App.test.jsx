/* global it, expect */

import React from 'react';
import renderer from 'react-test-renderer';
import App from '../client/src/components/App';

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();

  expect(tree).toMatchSnapshot();
});
