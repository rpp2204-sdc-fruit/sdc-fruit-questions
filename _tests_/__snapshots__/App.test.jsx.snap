// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="topbar"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium menu-icon css-i4bv87-MuiSvgIcon-root"
      data-testid="MenuIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
    <div
      className="atelier"
    >
      ATELIER
    </div>
    <div
      className="search-cart"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium search-icon css-i4bv87-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ShoppingCartIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </div>
  </div>
  <div
    id="reviews-body"
  >
    <div
      id="reviews-module"
    >
      <h2
        id="ratings-reviews"
      >
         Ratings and Reviews 
      </h2>
      <div
        id="ratings-breakdown"
      >
        <h3
          hidden={true}
        >
          Rating's Breakdown
        </h3>
        <div
          id="average-rating"
        >
          <div
            id="average"
          >
            0
          </div>
          <div
            id="breakdown-stars"
          />
        </div>
        <div
          id="review-count"
        >
          out of 0 ratings
        </div>
        <div
          id="recommend-percent"
        >
          0% of reviewers recommend this product
        </div>
        <div
          id="ratings-graph"
        >
          <div />
        </div>
        <div>
          <h3
            hidden={true}
          >
            Product Breakdown
          </h3>
        </div>
      </div>
      <div
        id="reviews-list"
      >
        <h3
          hidden={true}
        >
          Reviews List
        </h3>
        <div
          id="review-sort"
        >
          <div
            id="sort-text"
          >
            0 reviews, sorted by
          </div>
          <div
            id="sort-dropdown"
          >
            relevance
            <div
              id="sort-dropdown-content"
            >
              <ul
                className="review-sort-types"
              >
                <li
                  className="relevant"
                  onClick={[Function]}
                >
                  relevance
                </li>
                <li
                  className="newest"
                  onClick={[Function]}
                >
                  newest
                </li>
                <li
                  className="helpful"
                  onClick={[Function]}
                >
                  helpful
                </li>
              </ul>
            </div>
            <div
              id="sort-icon"
            >
              <i
                className="fa-regular fa-angle-down fa-lg"
              />
            </div>
          </div>
        </div>
        <div
          id="reviews"
        >
          <div
            className="review"
          >
            <h4
              hidden={true}
            >
              Review
            </h4>
            <div
              className="review-tile-top-bar"
            >
              <div
                className="stars"
              >
                <i
                  className="fak fa-star-thin star"
                />
                <i
                  className="fak fa-star-thin star"
                />
                <i
                  className="fak fa-star-thin star"
                />
                <i
                  className="fak fa-star-thin star"
                />
                <i
                  className="fak fa-star-thin star"
                />
              </div>
              <div
                className="date"
              />
              <div
                className="username"
              >
                ,
              </div>
            </div>
            <div
              className="summary"
            />
            <div
              className="body"
            />
            <div
              className="photos"
            />
            <div />
            <div
              className="helpfulness"
            >
              <div
                id="helpful-text"
              >
                <span
                  className="review-helpfulness"
                >
                  Helpful?
                </span>
                <span
                  className="helpful-yes"
                  onClick={[Function]}
                >
                  Yes 
                </span>
                <span
                  className="review-helpfulness"
                >
                  ()
                </span>
                <span
                  className="review-helpfulness"
                >
                  |
                </span>
                <span
                  className="review-report"
                  onClick={[Function]}
                >
                  Report
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        id="main-buttons"
      >
        <button
          className="reviews-btn"
          onClick={[Function]}
        >
          ADD A REVIEW +
        </button>
        <button
          className="reviews-btn"
          id="more-reviews"
          onClick={[Function]}
        >
          MORE REVIEWS
        </button>
      </div>
    </div>
  </div>
  <div
    id="QandA-main"
  >
    <div
      id="QandAtop"
    >
      <h1>
        Questions & Answers
      </h1>
      <div
        className="search"
      >
        <div
          className="searchInputs"
        >
          <input
            onChange={[Function]}
            placeholder="Have a question? Search for answers..."
            type="text"
            value=""
          />
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-magnifying-glass searchInputs"
            data-icon="magnifying-glass"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={{}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M500.3 443.7l-119.7-119.7c27.22-40.41 40.65-90.9 33.46-144.7C401.8 87.79 326.8 13.32 235.2 1.723C99.01-15.51-15.51 99.01 1.724 235.2c11.6 91.64 86.08 166.7 177.6 178.9c53.8 7.189 104.3-6.236 144.7-33.46l119.7 119.7c15.62 15.62 40.95 15.62 56.57 0C515.9 484.7 515.9 459.3 500.3 443.7zM79.1 208c0-70.58 57.42-128 128-128s128 57.42 128 128c0 70.58-57.42 128-128 128S79.1 278.6 79.1 208z"
              fill="currentColor"
              style={{}}
            />
          </svg>
        </div>
        <div
          className="dataResult"
        />
      </div>
      <div
        data-testid="test-questions"
        id="question-list"
      />
      <div
        id="questions-buttons"
      >
        <button
          onClick={[Function]}
        >
           Add Question + 
        </button>
      </div>
    </div>
  </div>
</div>
`;
